lib/create_send_rails/deliverer.rb

```diff
@@ -2,29 +2,60 @@ module CreateSendRails
   class Deliverer
     attr_accessor :settings
 
+    SAILTHRU = 'sailthru'.freeze
+    CREATSEND = 'creatsend'.freeze
+
     def initialize(values)
       self.settings = {}.merge(values)
     end
 
     def deliver!(mail)
       @mail = mail
 
+      delivery_system == SAILTHRU ? send_via_sailthru : send_via_createsend
+    end
+
+    private
+
+    attr_reader :mail
+
+    def sailthru_auth
+      settings.dup
+    end
+
+    def send_via_createsend
       smart_email = ::CreateSend::Transactional::SmartEmail.new(auth, smart_email_id)
```

Comment:
I believe `auth` here will throw undefined error, since we removed `def auth` and replaced it with `sailthru_auth` no?

Reply:
right indeed